Rakefile

```
@@ -68,28 +54,49 @@ parsed_files = PARSER_FILES.map do |parser_file|
     puts "Generating #{parsed_file}..."
     case ext
     when '.ry' # need racc
-      racc = Gem.bin_path 'racc', 'racc'
       rb_file = parser_file.gsub(/\.ry\z/, ".rb")
-      ruby "#{racc} -l -E -o #{rb_file} #{parser_file}"
+      sh "bundle exec racc -l -E -o #{rb_file} #{parser_file}"
       File.open(rb_file, 'r+') do |f|
         newtext = "# frozen_string_literal: true\n#{f.read}"
         f.rewind
         f.write newtext
       end
     when '.kpeg' # need kpeg
-      kpeg = Gem.bin_path 'kpeg', 'kpeg'
       rb_file = parser_file.gsub(/\.kpeg\z/, ".rb")
-      ruby "#{kpeg} -fsv -o #{rb_file} #{parser_file}"
+      sh "bundle exec kpeg -fsv -o #{rb_file} #{parser_file}"
       File.write(rb_file, File.read(rb_file).gsub(/ +$/, '')) # remove trailing spaces
     end
   end
 
   parsed_file
 end
 
-task "#{path}.gem" => package_parser_files
+RuboCop::RakeTask.new(:format_generated_files) do |t|
+  t.options = parsed_files + ["--config=.generated_files_rubocop.yml"]
+end
+
 desc "Generate all files used racc and kpeg"
-task :generate => parsed_files
+task generate: [*parsed_files, "format_generated_files:autocorrect"]
+
+desc "Verify that generated parser files are up to date"
+task verify_generated: :generate do
+  # Check if there are any uncommitted changes to the generated files
+  parsed_files.each do |file|
+    unless File.exist?(file)
+      abort "Generated file #{file} does not exist!"
+    end
+  end
+
+  diff_output = `git diff --exit-code #{parsed_files.join(' ')} 2>&1`
+  unless $?.success?
+    puts "Generated parser files are out of date!"
+    puts "Please run 'rake generate' and commit the changes."
+    puts "\nDifferences found:"
```

No argument `puts` may be more (a bit) readble than `\n`: 

```suggestion
    puts
    puts "Differences found:"
```